editoast/src/core/pathfinding.rs

```diff
@@ -15,7 +15,7 @@ pub struct PathfindingRequest {
     infra: i64,
     expected_version: String,
     waypoints: PathfindingWaypoints,
-    rolling_stocks: Vec<RollingStockForm>,
```

Comment:
If I understand correctly you changed that because you don't want to send a `locked` key that Core doesn't expect. But by replacing it by `RollingStock` you send the `id`, that Core doesn't expect either.

It works for now because Core simply ignores unknown keys but relying on that is extremely fragile imho.

My suggestion would be to extract the rolling stock fields that editoast and core share (ie: everything in `RollingStockForm` but `locked` and `metadata`) and move them in another struct (let's say `RollingStockCommon` but you can change the name), and include it in both `RollingStock` and `RollingStockForm`:

```rust
struct RollingStockForm {
    #[serde(flatten)]
    pub common: RollingStockCommon,
    pub locked: Option<bool>,
    pub metadata: RollingStockMetadata,
}

struct RollingStock {
    pub id: i64,
    #[serde(flatten)]
    pub common: RollingStockCommon,
}
```

Reply:
Wouldn't it be simpler to just move id to `RollingStockFull` ? i.e.:
```
struct RollingStockFull {
    pub id: i64,
    #[serde(flatten)]
    pub rolling_stock: RollingStock,
    pub locked: Option<bool>,
    pub metadata: RollingStockMetadata,
}

struct RollingStock {
    ... (without id)
}
```

Reply:
Indeed, since the views now all work with RollingStockFull. But then I think the naming is not ideal, what dot you think about something like that?

```rust
struct RollingStockCommon { // ex RollingStock
    ... (without id and metadata)
}

struct RollingStock { // ex RollingStockFull
    pub id: i64,
    #[serde(flatten)]
    pub common: RollingStockCommon,
    pub locked: Option<bool>,
    pub metadata: RollingStockMetadata,
}

struct RollingStockForm {
    #[serde(flatten)]
    pub common: RollingStockCommon,
    pub locked: Option<bool>,
    pub metadata: RollingStockMetadata,
}
```

Reply:
Yup that works ;)